File: Code/Color_LED/ColorLED.py
# -*- coding:UTF-8 -*-

import RPi.GPIO as GPIO
import time

#Definition of RGB module pin
LED_R = 22
LED_G = 27
LED_B = 24

#Set the GPIO port to BCM encoding mode.
GPIO.setmode(GPIO.BCM)

#Definition of Ultrasonic Module Pins
EchoPin = 0
TrigPin = 1

#Definition of servo pin
ServoPin = 23

#RGB pins are initialized into output mode
global pwm_ENA
global pwm_ENB
global pwm_servo
GPIO.setup(LED_R, GPIO.OUT)
GPIO.setup(LED_G, GPIO.OUT)
GPIO.setup(LED_B, GPIO.OUT)
GPIO.setup(ServoPin, GPIO.OUT)
pwm_servo = GPIO.PWM(ServoPin, 50)
pwm_servo.start(0)


#The servo rotates to the specified angle
def servo_appointed_detection(pos):
    for i in range(18):
        pwm_servo.ChangeDutyCycle(2.5 + 10 * pos/180)


#Display 7 color LED
try:
    while True:
        # Red for 1 Second
        GPIO.output(LED_R, GPIO.HIGH)
        GPIO.output(LED_G, GPIO.LOW)
        GPIO.output(LED_B, GPIO.LOW)
        servo_appointed_detection(0)
        time.sleep(1)


        # Green for 1 Second
        GPIO.output(LED_R, GPIO.LOW)
        GPIO.output(LED_G, GPIO.HIGH)
        GPIO.output(LED_B, GPIO.LOW)
        servo_appointed_detection(180)
        time.sleep(1)


        # Blue for 1 Second
        GPIO.output(LED_R, GPIO.LOW)
        GPIO.output(LED_G, GPIO.LOW)
        GPIO.output(LED_B, GPIO.HIGH)
        servo_appointed_detection(90)
        time.sleep(1)

except:
    print "except"


GPIO.cleanup()

File: Code/Color_LED/ColorLED.c
#include <wiringPi.h>

#define ON 1     
#define OFF 0    

//Definition of Pin
int LED_R = 3;           //LED_R is connected to  wiringPi port 3 of Raspberry pi
int LED_G = 2;           //LED_G is connected to  wiringPi port 2 of Raspberry pi
int LED_B = 5;           //LED_B is connected to  wiringPi port 5 of Raspberry pi

void color_led(int v_iRed, int v_iGreen, int v_iBlue)
{
  v_iRed == ON ? digitalWrite(LED_R, HIGH): digitalWrite(LED_R, LOW);
 
  v_iGreen == ON ? digitalWrite(LED_G, HIGH) : digitalWrite(LED_G, LOW);

  v_iBlue == ON ? digitalWrite(LED_B, HIGH) : digitalWrite(LED_B, LOW);
}

int main()
{
	wiringPiSetup();
	
	//Initialize the RGB IO as the output mode
	pinMode(LED_R, OUTPUT);
	pinMode(LED_G, OUTPUT);
	pinMode(LED_B, OUTPUT);
	
	while (1)
	{                        //   LED_R   LED_G    LED_B
    color_led(ON, OFF, OFF); //   1        0        0
    delay(3000);
    /// This configuration produces a steady Red Light.

    color_led(OFF, ON, OFF); //   0        1        0
    delay(2000);
    /// This configuration produces a steady Green Light.

   /// color_led(OFF, OFF, ON); //   0        0        1
   /// delay(1000);
   /// color_led(ON, ON, OFF);  //   1        1        0
   /// delay(1000);
   /// color_led(ON, OFF, ON);  //   1        0        1
  ///  delay(1000);
  ///  color_led(OFF, ON, ON);  //   0        1        1
   /// delay(1000);
  ///  color_led(ON, ON, ON);   //   1        1        1

	}
   return 0;	
}